Freshli/Languages/Ruby/RubyGemsRepository.cs

```diff
@@ -10,11 +10,16 @@ public class RubyGemsRepository : IPackageRepository {
     private IDictionary<string, IList<IVersionInfo>> _packages =
       new Dictionary<string, IList<IVersionInfo>>();
 
-    private IEnumerable<IVersionInfo> GetReleaseHistory(string name) {
+    private IEnumerable<IVersionInfo> GetReleaseHistory(
+      string name,
+      bool includePreReleaseVersions)
+    {
       try {
-
-        if (_packages.ContainsKey(name)) {
-          return _packages[name];
+        var keySuffix =
+          includePreReleaseVersions ? "-with-pre-releases" : "";
+        var key = $"{name}{keySuffix}";
+        if (_packages.ContainsKey(key)) {
+          return _packages[key];
         }
 
         var url = $"https://rubygems.org/gems/{name}/versions";
```

Comment:
@RepoCorp Yes, we used to use the API, and it would be cleaner, but unfortunately, it doesn't return yanked versions (see issue #50) which results in a lot of skipped dependencies.